### ISSUE TITLE
SelectedTracks

### ISSUE THREAD

**Issue by gizomo** [TheWidlarzGroup/react-native-video#3287]
# Bug

The latest changes to the player are not compatible with the previous API.
There is no method in Video.tsx for selecting video tracks.
In methods for selecting audio and text tracks, the fields of an selectedTrack now depend on its type (value/index). Changing audo track causes the sound to drop out.

There are no changes at Documentation related to selectedTracks too.

**Comment by freeboub** [comment]
@gizomo Thank you for the report.
@KrzysztofMoch can you have a look please ?
I confirm : selectedVideoTrack is missing in Video.tsx

Not sure about the point about selectedAudioTracks, do you have the issue on android ios or both ?

**Comment by gizomo** [comment]
At first comment I've mistaken. But problem exists.
Video.tsx contains this code where object can have **selectedAudioType** and **value/index**:
```
const _selectedAudioTrack = useMemo(() => {
  if (!selectedAudioTrack) {
    return;
  }
  if (typeof selectedAudioTrack?.value === 'number') {
    return {
      selectedAudioType: selectedAudioTrack?.type,
      index: selectedAudioTrack?.value,
    };
  }
  return {
    selectedAudioType: selectedAudioTrack?.type,
    value: selectedAudioTrack?.value,
  };
}, [selectedAudioTrack]);
```
But code for android Exoplayer waits only **type** and **value**
```
private static final String PROP_SELECTED_AUDIO_TRACK = "selectedAudioTrack";
private static final String PROP_SELECTED_AUDIO_TRACK_TYPE = "type";
private static final String PROP_SELECTED_AUDIO_TRACK_VALUE = "value";
```
The same goes for selectedTextTrack
```
private static final String PROP_SELECTED_TEXT_TRACK = "selectedTextTrack";
private static final String PROP_SELECTED_TEXT_TRACK_TYPE = "type";
private static final String PROP_SELECTED_TEXT_TRACK_VALUE = "value";
```

**Comment by gizomo** [comment]
As long as selectedAudioTrack is set to undefined, sound works. But as soon as you transfer any object to this prop, sound drops out.

**Comment by gizomo** [comment]
For IOS SelectedTrackCriteria also waits only **type** and **value**

**Comment by KrzysztofMoch** [comment]
Okay, @gizomo thanks for investigation 🙏
@freeboub looks like we need to update JS or Native side - lets discuss 
I will mark this as critical for release

**Comment by gizomo** [comment]
What was wrong with the previous implementation?

Why not always set an object with **type** and **value** at _selectedAudioTrack and _selectedTextTrack as it was before?

Now the declared types for selectedAudioTrack and selectedTextTrack don't even match its implementation at functions.
![Screenshot_20231011_142136](https://github.com/react-native-video/react-native-video/assets/2512011/d23da2fc-0f67-4035-a5cb-266ef4cf59b7)
![Screenshot_20231011_142205](https://github.com/react-native-video/react-native-video/assets/2512011/e68e8e02-811d-4e23-be24-aca88c1c2dfb)

**Comment by KrzysztofMoch** [comment]
@gizomo you right, i have made mistake while adding types - created PR #3288 with fix. 
Cloud you test it ? Thanks for your help 🙌

**Comment by freeboub** [comment]
Sorry, I already merge the fix, I was looking good on my side.
@gizomo your feedback is anyway welcome !
I close the ticket, will be available in 6.0.0-alpha.9
